packages/rich-text/src/plugins/shared/LinkedInlineWrapper.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+
+import tokens from '@contentful/f36-tokens';
+import { EntryLink, ResourceLink } from '@contentful/field-editor-reference';
+import { css } from 'emotion';
+
+import { IS_CHROME } from '../../helpers/environment';
+import { Element, RenderElementProps } from '../../internal/types';
+
+const styles = {
+  icon: css({
+    marginRight: '10px',
+  }),
+
+  root: css({
+    display: 'inline-block',
+    margin: `0 ${tokens.spacing2Xs}`,
+    fontSize: 'inherit',
+    span: {
+      userSelect: 'none',
+    },
+  }),
+};
+
+const isResourceLink = (link: EntryLink | ResourceLink): link is ResourceLink =>
+  !!(link as ResourceLink).sys.urn;
+
+type LinkedInlineWrapperProps = React.PropsWithChildren<{
+  attributes: Pick<RenderElementProps, 'attributes'>;
+  card: JSX.Element;
+  element: Element & {
+    data: {
+      target: ResourceLink | EntryLink;
+    };
+  };
+}>;
+
+export function LinkedInlineWrapper({
+  attributes,
+  card,
+  children,
+  element,
+}: LinkedInlineWrapperProps) {
+  const link = element.data.target;
```

Comment:
Could we only pass the `target` down if that's all we need? Or will we need more later on?